flang/lib/Lower/ConvertVariable.h

```diff
@@ -77,13 +78,22 @@ void mapCallInterfaceSymbols(AbstractConverter &,
                              const Fortran::lower::CallerInterface &caller,
                              SymMap &symMap);
 
+// TODO: consider having an initial expression analysis saved in the PFT and
+// creating the symbol map once when creating the global op instead of dealing
+// with this at the expression level.
```

Comment:
I don't understand this comment. What is it referring to?

Reply:
It's a TODO for me that I do not like the way we creating variable address_of in globalOp "at the use point" when meeting the sub-expression instead of when starting lowering the globalOp (That is the reason why we need this new function, otherwise, using `genExprAddr` should just work, even inside gloablOp). I clarified the comment.